core/commands/dag/dag.go

```diff
@@ -343,4 +355,53 @@ The output of blocks happens in strict DAG-traversal, first-seen, order.
 
 		return err
 	},
+	PostRun: cmds.PostRunMap{
+		cmds.CLI: func(res cmds.Response, re cmds.ResponseEmitter) error {
+
+			var showProgress bool
+			val, specified := res.Request().Options[progressOptionName]
+			if !specified {
+				// default based on TTY availability
+				errStat, _ := os.Stderr.Stat()
+				if 0 != (errStat.Mode() & os.ModeCharDevice) {
+					showProgress = true
+				}
+			} else if val.(bool) {
+				showProgress = true
+			}
+
+			// simple passthrough
+			if !showProgress {
+				return cmds.Copy(re, res)
+			}
+
+			// We will be supplied one or more readers, simply wrap them with a progress bar
+			for {
+				v, err := res.Next()
+				if err == io.EOF {
+					return re.Close()
+				} else if err != nil {
+					return re.CloseWithError(err)
+				}
+
+				r, ok := v.(io.Reader)
+				if !ok {
+					// some sort of encoded response, this should not be happening
+					return errors.New("unexpected non-stream passed to PostRun: can't happen")
+				}
+
+				if err := func() error {
+					bar := pb.New64(0).SetUnits(pb.U_BYTES)
```

Comment:
How does this look when adding multiple files? It would be nice to see:

```
Exported Qm...
Exported Qm...
Exported Qm...
[ progress ]
```

Instead of having dead progress bars in between. But it's not a _huge_ issue.

Reply:
> How does this look when adding multiple files? 

We can only export one root/DAG ( single CID spec, no selectors ). The exporter only sees "bytes written" into a single stream on STDOUT. No files are created.

Reply:
Oh, yeah. I knew that...

Reply:
I think I see what you mean... let me rewire this / rebase on top of the approved export.

Reply:
No you are right - there may be a reason for the server to emit multiple read-ables for some reason. So having the bar outside of the loop make sense. I will redo it.

Reply:
That shouldn't happen... and I'm not sure what we'd be able to do about it. Would we just concatenate them?

Reply:
Yep, concat.

Alternatively we can throw saying "unexpected multipart export..." or somesuch.

*OR* we can leave the multi-bar spawning as-is ( no changes to PR ), then there will eb a visual cue "something went wrong"

But if you say "this shouldn't happen" - erroring is what I prefer.

Reply:
I'd error.

Reply:
Addressed as https://github.com/ipfs/go-ipfs/pull/7037/files#diff-15172926c4147422f472139313467179R388-R396

There is more code on the closer, as that became a can of worms, but it will go in the other PR once this is folded in ( I promise it will make sense in a bit )

Waiting for CI to run here, and will merge this into the other branch.